File: services/frontend/src/lib/api/applications/create-application-from-eakte.ts
export async function createApplicationFromEakte(applicantID: string, eakteID: string, schoolDegreeID: string, educationStartDate: Date, educationEndDate: Date, classLevel: string, labelIDs: string[] = []): Promise<boolean> {
  try {
    const requestBody: ApplicationCreateFromEakteInsertModel = {
      applicantID,
      eakteID,
      schoolDegreeID,
      educationStartDate: educationStartDate.toISOString(),
      educationEndDate: educationEndDate.toISOString(),
      classLevel,
      labelIDs
    };

    const response = await fetch('/api/v1/applications/eakte-new-application', {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    if (!response.ok) {
      throw new Error('Failed to create application from E-Akte');
    }

    return true;
  } catch (error) {
    console.error('Error creating application from E-Akte:', error);
    return false;
  }
}

File: services/backend/src/resources/applications/http/create_application_from_eakte.go
package http

import (
	"time"

	serviceModel "github.com/dcssoftware/bafoeg-manager/src/resources/applications/service/models"
	"github.com/gofiber/fiber/v3"
	"github.com/google/uuid"
)

// CreateApplicationFromEakteRequest represents the request body for creating an application from eAkte
type CreateApplicationFromEakteRequest struct {
	ApplicantID        uuid.UUID   `json:"applicantID"`
	EakteID            uuid.UUID   `json:"eakteID"`
	SchoolDegreeID     uuid.UUID   `json:"schoolDegreeID"`
	EducationStartDate time.Time   `json:"educationStartDate"`
	EducationEndDate   time.Time   `json:"educationEndDate"`
	ClassLevel         string      `json:"classLevel"`
	LabelIDs           []uuid.UUID `json:"labelIDs"`
}

func (h *ApplicationsHandler) CreateApplicationFromEakte(c fiber.Ctx) error {
	// Parse request body
	var req CreateApplicationFromEakteRequest
	if err := c.Bind().Body(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).SendString("Invalid request body: " + err.Error())
	}

	applicationID, revisionID, mappingID, applicationErr := h.service.InsertApplicationFromEakte(nil, serviceModel.ApplicationFromEakteInsertModel{
		ClassLevel:         req.ClassLevel,
		EakteAkteID:        req.EakteID,
		LabelIDs:           req.LabelIDs,
		ApplicantID:        req.ApplicantID,
		SchoolWithDegreeID: req.SchoolDegreeID,
		EducationStart:     req.EducationStartDate,
		EducationEnd:       req.EducationEndDate,
	})

	if applicationErr != nil {
		status, message := applicationErr.HTTPError()
		return c.Status(status).SendString(message)
	}

	return c.JSON(map[string]any{
		"success":          true,
		"application_id":   applicationID,
		"base_revision_id": revisionID,
		"mapping_id":       mappingID,
	})
}
